### ISSUE TITLE
Error applying new propagation method to Neo Blocks

### ISSUE THREAD

**Issue by MattWilcox** [spicywebau/craft-neo#434]
#### Description

Updated Craft and all plugins. Running totally current software.

Went to a Neo field and changed the propagation method from `save blocks to other sites in the same site group` to `Only dave blocks to the site they were created in`

Failed task results...

```
Applying new propagation method to Neo blocks
ID	5086
Status	Failed
Error	Invalid owner ID: 55584
Progress	3% (110 of 3,529)
Time to reserve	300 seconds
Priority	1024
Pushed at	17 Mar 2021, 12:42:29
Updated at	17 Mar 2021, 12:42:42
Failed at	17 Mar 2021, 12:42:42
Job Data
{
    "elementType": "benf\\neo\\elements\\Block",
    "criteria": {
        "fieldId": "164"
    },
    "description": "Applying new propagation method to Neo blocks"
}
```

#### Other information
- Neo version: 2.9.3
- Craft version: 3.6.10
- (if multi-site) What is the affected Neo field's propagation method: Only dave blocks to the site they were created in
- Is eager-loading used? Yes

[composer.zip](https://github.com/spicywebau/craft-neo/files/6156452/composer.zip)
[error-logs.zip](https://github.com/spicywebau/craft-neo/files/6156454/error-logs.zip)

**Comment by ttempleton** [comment]
If you use a nested Matrix (or Super Table) field in a Neo block, could you please check the `neoblocks` and `matrixblocks` (and `supertableblocks`) and confirm which table has a row with `ownerId = 55584`?  If it is a Matrix / Super Table block, could you please confirm the propagation method of that field?

**Comment by ttempleton** [comment]
Thanks for that; unfortunately I'm still unable to reproduce this.  If possible, could you please send your database backup to plugins@spicyweb.com.au and we'll take a look at what's happening.

**Comment by MattWilcox** [comment]
I've emailed in with a link to the DB, though #436 is the more pressing bug at the moment! Thanks again for looking into all this...

**Comment by ttempleton** [comment]
Thanks for sending that, @MattWilcox; turned out to be a bug with the Neo block query grabbing the wrong structure ID for a block in some cases, which caused both this and #436, so they should both be fixed in Neo 2.9.5.

**Comment by MattWilcox** [comment]
Nice one, thanks!
